posthog/async_migrations/migrations/0004_replicated_schema.py

```diff
@@ -147,27 +163,33 @@ def replicated_table_operations(self, table: TableMigrationData):
 
         yield AsyncMigrationOperation(
             fn=lambda _: self.rename_tables(
-                table.name, table.tmp_table_name, table.renamed_table_name, table.backup_table_name
+                [table.name, table.backup_table_name], [table.tmp_table_name, table.renamed_table_name],
             ),
             rollback_fn=lambda _: self.rename_tables(
-                table.backup_table_name,
-                table.renamed_table_name,
-                table.tmp_table_name,
-                table.name,
-                skip_if_backup_exists=True,
+                [table.renamed_table_name, table.tmp_table_name],
+                [table.backup_table_name, table.name],
+                verification_table=table.backup_table_name,
             ),
         )
 
-        if table.materialized_view_name is not None:
+    def finalize_table_operations(self, table: TableMigrationData):
+        # NOTE: Relies on IF NOT EXISTS on the query
+        if isinstance(table, ShardedTableMigrationData):
+            for table_name, create_table_query in table.extra_tables.items():
+                yield AsyncMigrationOperationSQL(sql=create_table_query, rollback=f"DROP TABLE IF EXISTS {table_name}")
+
+        if isinstance(table, ShardedTableMigrationData) and table.materialized_view_name is not None:
+            yield AsyncMigrationOperationSQL(sql=f"DROP TABLE IF EXISTS {table.materialized_view_name}", rollback=None)
```

Comment:
nit: could we use `DROP VIEW` for materialized views? Literally won't change a thing, just helps with readability amongst all the drop statements

Reply:
Migrations also use `DROP TABLE` though.

Reply:
yeah that's true